[iOS] SearchHandler overlaps title and title view - fix

kubaflo opened the pull request:
<!-- Please let the below note in for people that find this PR -->
> [!NOTE]
> Are you waiting for the changes in this PR to be merged?
> It would be very helpful if you could [test the resulting artifacts](https://github.com/dotnet/maui/wiki/Testing-PR-Builds) from this PR and let us know in a comment if this change resolves your issue. Thank you!

<!--
!!!!!!! MAIN IS THE ONLY ACTIVE BRANCH. MAKE SURE THIS PR IS TARGETING MAIN. !!!!!!! 
-->

### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/30771

|Before|After|
|--|--|
|<video src="https://github.com/user-attachments/assets/f553f3ae-0dc1-41a2-9f56-6b6b606a0a33" width="300px"></video>|<video src="https://github.com/user-attachments/assets/3eb260ec-a210-4ea4-9493-d71927d849b0" width="300px"></video>|

kubaflo committed:
Added a UI Test

kubaflo committed:
Added snapshots

PureWeen commented:
/azp run

jsuarezruiz reviewed (commented):
@kubaflo Could you rebase and fix the conflict?

kubaflo committed:
Handle SearchBar editing stopped event on iOS

Added support for the OnEditingStopped event on the SearchBar in ShellPageRendererTracker for iOS. This ensures the search controller is deactivated when editing stops and properly unsubscribes from the event during disposal to prevent memory leaks.
